### ISSUE TITLE
deepseekocr: hard-coded BOS and tokenizer whitespace loss cause hallucinated page preamble with jina-ocr-v1 (DeepSeek-OCR fine-tune)

### ISSUE THREAD

**Issue by cjnuk** [Blaizzy/mlx-vlm#2300]
## Summary

With [`jinaai/jina-ocr-v1`](https://huggingface.co/jinaai/jina-ocr-v1), a DeepSeek-OCR fine-tune with the same architecture, the `deepseekocr` prompt path gives the model a different token sequence from the reference HF processor. The model then invents text at the top of the page. Two things cause this:

1. **The BOS token is hard-coded.** `models/deepseekocr/processing_deepseekocr.py` always prepends `bos_id = 0` (`tokenized_str = [bos_id] + tokenized_str`, lines ~486–487 on `main`), whatever `add_special_tokens` or the tokenizer's `add_bos_token` says. Original DeepSeek-OCR expects BOS. jina-ocr-v1's reference processor does not add it, and its `tokenizer_config.json` has `"add_bos_token": false`.
2. **Whitespace is lost with this tokenizer under transformers 5.x.** The checkpoint's `tokenizer_config.json` declares `"tokenizer_class": "LlamaTokenizerFast"`. Loaded through mlx-vlm with transformers 5.12.1 or 5.17.0, the tokenizer is rebuilt with a `Metaspace` pre-tokenizer instead of the byte-level `Sequence` from `tokenizer.json`, so leading spaces and newlines disappear:

```
text: "<|User|>:\nTranscribe the provided document"
mlx-vlm tokenizer : ['<|User|>', ':', 'Trans', 'crib', 'ethe', 'provided', 'document']
tokenizer.json    : ['<|User|>', ':\n', 'Trans', 'cribe', ' the', ' provided', ' document']
```

Changing the class to `PreTrainedTokenizerFast` in the checkpoint fixes (2). Only a code change can fix (1).

## Evidence

Environment: mlx-vlm 0.6.3, mlx 0.32.2, mlx-lm 0.31.3, macOS on an Apple M5 Max. Checkpoint `jinaai/jina-ocr-v1@904f815d`, repackaged for mlx-vlm (`model_type: deepseekocr`, `mtp_module.*` draft-head tensors dropped, and `candidate_resolutions`/`patch_size`/`downsample_ratio` added to `processor_config.json`). All retained weights are byte-identical BF16.

The test is one scanned book page (1052×1395), compared with the output of the reference `transformers` implementation (4.57.3, CPU, greedy):

| Input ids | Output vs HF reference |
|---|---|
| mlx-vlm prompt path, default tokenizer | invented text at the top (e.g. `.)\n\nQUESTION 1\n\n`), otherwise near-identical |
| tokenizer class fixed, BOS still prepended | invented `):\n` at the top |
| **reference input_ids passed via `generate(..., input_ids=...)`** | **byte-identical** (752 tokens) |

Pixel preprocessing matches the reference (same 2×3 crops of 640 plus a 1024 global view, mean |Δ| ≈ 7e-5), so the difference comes entirely from the text tokens.

## Suggested fix

In `DeepseekOCRProcessor`, prepend BOS only when configured to, for example by honouring `add_special_tokens` (the argument `prepare_inputs` already passes), the processor's existing `add_special_token` attribute, or the tokenizer's `add_bos_token`. Keep the current default so original DeepSeek-OCR checkpoints behave as before. For (2), it may be worth preferring `tokenizer.json` (`PreTrainedTokenizerFast`) when a checkpoint ships one.

I'm happy to test a patch against this checkpoint.
